pkg/gateway/services/stub.go

```diff
@@ -161,6 +161,14 @@ func (gws *GatewayService) DeployStub(ctx context.Context, in *pb.DeployStubRequ
 
 	version := uint(1)
 	if lastestDeployment != nil {
+		if stub.Type == types.StubType(types.StubTypeScheduledJobDeployment) {
+			if err := gws.stopDeployments([]types.DeploymentWithRelated{*lastestDeployment}, ctx); err != nil {
```

Comment:
Should this be after `deployment, err := gws.backendRepo.CreateDeployment` in case that fails?

Reply:
What if CreateDeployment succeeds and stopDeployments fails? Someone could keep trying and create more Deployments.

Reply:
`stopDeployments` also only returns an error if backendRepo.UpdateDeployment() fails. Errors are ignored for all other calls in that function.

I think this should be fine.